lib/services/northBound/contextServer.js

```diff
@@ -820,7 +820,7 @@ function handleQueryNgsi2(req, res, next) {
         getFunction(function handleFindDevice(error, innerDevice) {
             let deviceList = [];
             if (!innerDevice) {
-                callback(new errors.DeviceNotFound(contextEntity.id));
+                return callback(new errors.DeviceNotFound(contextEntity.id));
```

Comment:
I think this `return` is needed here. Otherwise, the execution flow will continue and `if (innerDevice.count)` will fail, as `innerDevice` would not have any field named `count`.

Reply:
You're right, thanks for the fix!

Reply:
NTC